Directory.Build.props

```diff
@@ -21,7 +21,28 @@
       <BUILDING_USING_DOTNET>true</BUILDING_USING_DOTNET>
   </PropertyGroup>
 
-  <Import Project="$(RepoRoot)/Directory.Build.props.user" Condition = "Exists('$(RepoRoot)/Directory.Build.props.user')" />
+  <Import Project="$(RepoRoot)/Directory.Build.props.user" Condition="Exists('$(RepoRoot)/Directory.Build.props.user')" />
+
+ <!-- To support source-build for the "next" version of .NET without forcing this repo to use it,
+      we can utilize arcade's $(NetCurrent) property, and set it ourselves in the arcade-less scenarios. 
+      For FCS solution, we set it to one for current dev cycle.
+      When used with arcade, and inserting/building for net8 branches, it will set it to net8.0,
+      for net9 branches to net9.0 and so on -->
+  <Choose>
+    <!-- Once we move to OOP in VS, and major dependants of FCS will move to netcore (not netstandard), we should also support $(NetPrevious) for all releases.
+         This will likely include FCS and FSharp.Core as well ash shipped products.
+         Right now, it only covers producs we ship (FSC and FSI), not NuGet packages. -->
+    <When Condition="'$(DotNetBuildFromSource)' == 'true' AND '$(DotNetBuildFromSourceFlavor)' == 'Product'">
```

Comment:
@ViktorHofer @mthalman will this be enough for SB? Or check for `Product` is redundant?

Reply:
This looks correct. This will ensure that NetCurrent is used when running source build in the context of the VMR (that's what `Product` means here). The other scenario would be running source build in the context of the fsharp repo. In that case, you would still want to use `net8.0` since that's what the repo is configured to use. And so this condition correctly gives you those two behaviors.

Reply:
> This looks correct. This will ensure that NetCurrent is used when running source build in the context of the VMR (that's what `Product` means here). The other scenario would be running source build in the context of the fsharp repo. In that case, you would still want to use `net8.0` since that's what the repo is configured to use. And so this condition correctly gives you those two behaviors.

How do I test the former?

Reply:
The changes would need to be applied to the [VMR](https://github.com/dotnet/dotnet) repo. I'll take care of that by creating a branch and running it through our CI.